engine/src/main/java/org/destinationsol/location/systems/LocationSystem.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.location.systems;
+
+import org.destinationsol.entitysystem.EventReceiver;
+import org.destinationsol.location.components.Angle;
+import org.destinationsol.location.components.Position;
+import org.destinationsol.location.components.Velocity;
+import org.destinationsol.location.events.AngleUpdateEvent;
+import org.destinationsol.location.events.PositionUpdateEvent;
+import org.destinationsol.location.events.VelocityUpdateEvent;
+import org.terasology.gestalt.entitysystem.entity.EntityRef;
+import org.terasology.gestalt.entitysystem.event.EventResult;
+import org.terasology.gestalt.entitysystem.event.ReceiveEvent;
+
+/**
+ * This system updates the location of an entity with a {@link Position}, {@link Angle}, or {@link Velocity} component
+ * when it receives a {@link PositionUpdateEvent}, {@link AngleUpdateEvent}, or {@link VelocityUpdateEvent}, respectively.
+ */
+public class LocationSystem implements EventReceiver {
+
+    @ReceiveEvent(components = Position.class)
+    public EventResult onPositionUpdate(PositionUpdateEvent event, EntityRef entity) {
+        if (entity.hasComponent(Position.class)) {
+            Position position = entity.getComponent(Position.class).get();
+            position.position = event.getPosition();
```

Comment:
Position position position.position. Hmm maybe there's a better name for some of these :thinking: 
Just thinking aloud, not the end of the world :+1:

Reply:
What do you think about positionComponent?

Reply:
If you're just talking about the local field then yeah that sounds good :+1:

Reply:
As in "positionComponent.position"